### DISCUSSION TITLE
isConnecting from useAccount hook stays true after closing web3modal

### DISCUSSION THREAD

**Discussion by Melancholism** [wevm/wagmi#2870] [Q&A]
I'm using wagmi with web3modal. When I open the w3m with its `open()` function and close it **without connecting my wallet**, the `isConnecting` constant from wagmi's `useAccount()` hook stays **true**. Why is this happening? Am I doing something wrong? or is it normal and the natural behavior of the hook?

```js
const { open } = useWeb3Modal();
const { isConnecting } = useAccount();

<button onClick={() => open()}>Open Modal</button>;
```
In the snippet above, before clicking the _Open Modal button_, `isConnecting` is **false**. But after opening the w3m and **closing** it, `isConnecting` gets and remains **true**.

**Comment by Melancholism** [comment]
@jxom 
can you please help me with this? sorry to ping and bother you.

**Comment by MutiatBash** [comment]
I’m also experiencing this issue when my app loads IsConnecting from useAccount stays true even when a connection to a wallet has not been made yet

**Comment by pedrus16** [comment]
Hello, I'm having the same issue.

As a workaround I check both `isConnecting && open` states from `useAccount()` and `useWeb3ModalState()` respectively.
